src/codeProfilerFunction.ts

```diff
@@ -2,22 +2,26 @@
 /* eslint-disable @typescript-eslint/naming-convention */
 'use strict';
 import * as vscode from 'vscode';
-const fs = require('fs');
-const os = require('os');
-const path = require('path');
-const dns = require('node:dns');
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as dns from 'dns';
 import axios from 'axios';
 import { Agent } from 'https';
-const util = require('util');
-const stream = require('stream');
 const extract = require('extract-json-from-string');
-const pipeline = util.promisify(stream.pipeline);
+const homedir = require('os').homedir();
```

Comment:
```suggestion
const tmpdir = require('os').tmpdir();
```
And then adjusting homedir below

Reply:
done